be/src/olap/rowset/segment_v2/segment_iterator.h

```diff
@@ -49,16 +49,19 @@ class SegmentIterator : public RowwiseIterator {
     Status next_batch(RowBlockV2* row_block) override;
     const Schema& schema() const override { return _schema; }
 private:
-    Status _init_short_key_range();
-    Status _prepare_seek();
-    Status _init_row_ranges();
-    Status _get_row_ranges_from_zone_map(RowRanges* zone_map_row_ranges);
-    Status _init_column_iterators();
-    Status _create_column_iterator(uint32_t cid, ColumnIterator** iter);
-
+    // calculate row ranges that fall into requested key ranges using short key index
+    Status _get_row_ranges_by_keys();
+    Status _prepare_seek(const StorageReadOptions::KeyRange& key_range);
     Status _lookup_ordinal(const RowCursor& key, bool is_include,
                            rowid_t upper_bound, rowid_t* rowid);
     Status _seek_and_peek(rowid_t rowid);
+
+    // calculate row ranges that satisfy requested column conditions using various column index
+    Status _get_row_ranges_by_column_conditions();
+    Status _get_row_ranges_from_zone_map(RowRanges* zone_map_row_ranges);
```

Comment:
add a get_row_ranges_by_conditions function to column_reader and move this function to column_reader.
so column reader will care about all the indexes processing including zone map/bf/bitmap.

Reply:
Good suggestion. I think I'll do the refactor later (combined with the PR to support lazy init of SegmentIterator and ColumnReader)